dockstore-webservice/src/main/java/io/dockstore/webservice/core/SourceFile.java

```diff
@@ -234,18 +234,18 @@ public void setFrozen(boolean frozen) {
         this.frozen = frozen;
     }
 
-    private static void checkPath(String path) {
+    private static synchronized void checkPath(String path) {
```

Comment:
From a performance standpoint, I think I preferred the volatile approach better, even though synchronized is correct. :) But no need to change, and this is correct.